File: ruby-resend-examples/examples/batch_send.rb
#!/usr/bin/env ruby
# frozen_string_literal: true

##
# Batch Email Sending Example
#
# Demonstrates sending multiple emails in a single API call.
#
# Key points:
# - Maximum 100 emails per batch
# - No attachments supported in batch
# - No scheduling supported in batch
#
# Usage: ruby examples/batch_send.rb

require "bundler/setup"
require "resend"
require "dotenv/load"

Resend.api_key = ENV.fetch("RESEND_API_KEY")

from_email = ENV.fetch("EMAIL_FROM", "Acme <onboarding@resend.dev>")
contact_email = ENV.fetch("CONTACT_EMAIL", "team@example.com")

begin
  # Batch send: multiple emails in one API call
  result = Resend::Batch.send([
    # Email 1: Confirmation to user
    {
      from: from_email,
      to: ["user@example.com"],
      subject: "We received your message",
      html: "<h1>Thanks for reaching out!</h1><p>We'll get back to you soon.</p>"
    },
    # Email 2: Notification to team
    {
      from: from_email,
      to: [contact_email],
      subject: "New contact form submission",
      html: "<h1>New message received</h1><p>From: user@example.com</p>"
    }
  ])

  puts "Batch sent successfully!"
  result["data"].each_with_index do |email, index|
    puts "Email #{index + 1} ID: #{email["id"]}"
  end
rescue StandardError => e
  puts "Error sending batch: #{e.message}"
  exit 1
end

File: python-resend-examples/examples/batch_send.py
"""
Batch Email Sending Example

Demonstrates sending multiple emails in a single API call.

Key points:
- Maximum 100 emails per batch
- No attachments supported in batch
- No scheduling supported in batch
- If one email fails validation, entire batch fails

Usage: python examples/batch_send.py

@see https://resend.com/docs/api-reference/emails/send-batch-emails
"""

import os
import resend
from dotenv import load_dotenv

load_dotenv()
resend.api_key = os.environ["RESEND_API_KEY"]


def main():
    """Send batch emails (e.g., for contact forms)."""
    from_email = os.environ.get("EMAIL_FROM", "Acme <onboarding@resend.dev>")
    contact_email = os.environ.get("CONTACT_EMAIL", "delivered@resend.dev")

    try:
        # Batch send: multiple emails in one API call
        result = resend.Batch.send([
            # Email 1: Confirmation to user
            {
                "from": from_email,
                "to": ["delivered@resend.dev"],
                "subject": "We received your message",
                "html": "<h1>Thanks for reaching out!</h1><p>We'll get back to you soon.</p>",
            },
            # Email 2: Notification to team
            {
                "from": from_email,
                "to": [contact_email],
                "subject": "New contact form submission",
                "html": "<h1>New message received</h1><p>From: delivered@resend.dev</p>",
            },
        ])

        print("Batch sent successfully!")
        for i, email in enumerate(result["data"]):
            print(f"Email {i + 1} ID: {email['id']}")

    except Exception as e:
        print(f"Error sending batch: {e}")
        exit(1)


if __name__ == "__main__":
    main()
